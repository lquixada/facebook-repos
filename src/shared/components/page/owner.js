import React from 'react'
import styled from 'styled-components'

import Menu from '../../containers/menu'
import {Footer} from '../footer'
import {Aside, Content, Section, SectionContainer} from '../grid'
import {flex} from '../../helpers'

export const OwnerPage = () => (
  <Section>
    <SectionContainer>
      <Aside>
        <Menu />
        <Footer />
      </Aside>
      <Content>
        <Wrapper>
          Choose a repository on the menu.
        </Wrapper>
      </Content>
    </SectionContainer>
  </Section>
)

const Wrapper = styled.div`
  ${flex.display()}
  ${flex.middle()} 
  ${flex.center()}
  height: 50rem;
  border: .7rem dashed #eee;
  border-radius: .6rem;
  font-size: 1.4rem;
  letter-spacing: -.1px;
`
