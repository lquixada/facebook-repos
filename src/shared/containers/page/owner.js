import React from 'react'
import {compose} from 'redux'
import {connect} from 'react-redux'
import {OwnerPage} from '../../components/page/owner'
import {provideHooks} from '../../helpers'
import * as actions from '../../actions'

const hooks = {
  fetch: ({params, dispatch}) => dispatch(actions.fetchPage({
    name: 'owner',
    owner: params.owner || 'facebook'
  }))
}

export class OwnerPageContainer extends React.Component {
  componentDidMount () {
    this.fetch()
  }

  componentDidUpdate (prevProps) {
    if (this.hasChanged(prevProps)) {
      this.fetch()
    }
  }

  fetch () {
    this.props.fetchPage({
      name: 'owner',
      owner: this.props.owner || 'facebook'
    })
  }

  hasChanged (prevProps) {
    return (
      prevProps.owner !== this.props.owner
    )
  }

  render () {
    return (
      <OwnerPage />
    )
  }
}

const mapStateToProps = (state, {match}) => ({
  owner: match.params.owner,
  repoName: match.params.repo
})
const composed = compose(provideHooks(hooks), connect(mapStateToProps, actions))
export default composed(OwnerPageContainer)
