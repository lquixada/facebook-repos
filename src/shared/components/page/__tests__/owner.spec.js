import React from 'react'
import {shallow} from 'enzyme'

import {OwnerPage} from '../owner'

describe('<OwnerPage />', () => {
  it('matches the snapshot', () => {
    const component = shallow(<OwnerPage />)
    expect(component).toMatchSnapshot()
  })
})
