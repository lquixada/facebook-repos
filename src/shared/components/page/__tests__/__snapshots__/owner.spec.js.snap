// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OwnerPage /> matches the snapshot 1`] = `
<styled.section>
  <styled.div>
    <styled.aside>
      <withRouter(Connect(MenuContainer)) />
      <Footer />
    </styled.aside>
    <styled.article>
      <styled.p>
        Choose a repository on the menu.
      </styled.p>
    </styled.article>
  </styled.div>
</styled.section>
`;
